epics/sync.js

```diff
@@ -0,0 +1,72 @@
+import { Observable } from 'rxjs';
+import { combineEpics } from 'redux-observable';
+import { LOCATION_CHANGE } from 'react-router-redux';
+import {
+	apiRequest,
+	apiSuccess,
+	apiError,
+	apiComplete,
+	locationSync,
+} from '../actions/syncActionCreators';
+import { activeRouteQueries$ } from '../util/routeUtils';
+import { fetchQueries } from '../util/fetchUtils';
+
+/**
+ * Navigation actions will provide the `location` as the payload, which this
+ * epic will use to collect the current Reactive Queries associated with the
+ * active routes.
+ *
+ * These queries will then be dispatched in the payload of `apiRequest`
+ * @param {Object} routes The application's React Router routes
+ * @returns {Function} an Epic function that emits an API_REQUEST action
+ */
+export const getNavEpic = routes => {
```

Comment:
Yeah - the only flag that this is a higher order function is the `get` prefix that isn't on the other epics, but hopefully it won't throw people off too much.